persist/fs/read.go

```diff
@@ -63,43 +66,159 @@ type reader struct {
 	blockSize      time.Duration
 
 	infoFdWithDigest           digest.FdWithDigestReader
-	indexFdWithDigest          digest.FdWithDigestReader
-	dataFdWithDigest           digest.FdWithDigestReader
 	digestFdWithDigestContents digest.FdWithDigestContentsReader
-	expectedInfoDigest         uint32
-	expectedIndexDigest        uint32
-	expectedDataDigest         uint32
-	expectedDigestOfDigest     uint32
-
-	unreadBuf   []byte
-	entries     int
-	entriesRead int
-	prologue    []byte
-	decoder     encoding.Decoder
-	digestBuf   digest.Buffer
-	bytesPool   pool.CheckedBytesPool
+
+	indexFd     *os.File
+	indexMmap   []byte
+	indexReader *decoderStream
```

Comment:
nit: maybe better to call it `indexDecoderStream` ?

Reply:
Sure thing.